sockeye/beam_search.py

```diff
@@ -677,7 +679,12 @@ def forward(self,
             target_dists, model_states, target_factors = self._inference.decode_step(best_word_indices,
                                                                                      model_states,
                                                                                      vocab_slice_ids)
-
+            # Do not generate <unk> token if prevent_unk is specified
```

Comment:
can we move this block into the `_Inference`  classes maybe? It would be good to keep the core beam search loop clean and high-level and move logic into sub-blocks. One reason is that it may make it easier in the future to convert these blocks into compiled computation graphs as much as possible.

Reply:
updated the code. I move the block into  `__Inference`